### DISCUSSION TITLE
Running colima with `--arch x86_64 --vm-type=vz --vz-rosetta` does not actually run Ubuntu VM on Rosetta

### DISCUSSION THREAD

**Discussion by maneesh29s** [abiosoft/colima#1150] [General]
I wanted to run the "guest" Ubuntu VM in x86_64 architecture, on rosetta.

At first I  ran `colima start -p ubuntu-arm --vm-type=vz --vz-rosetta`, and then ran `colima status`, it showed that architecture is `aarch64`. I thought it would be `x86_64` because Rosetta is used to emulate it on arm.

```
-> % colima status -p ubuntu-arm
INFO[0000] colima [profile=ubuntu-arm] is running using macOS Virtualization.Framework
INFO[0000] arch: aarch64
INFO[0000] runtime: docker
INFO[0000] mountType: virtiofs
INFO[0000] socket: unix:///Users/maneeshsutar/.colima/ubuntu-arm/docker.sock
```
In above case I could also see a `com.apple.Virtualization.VirtualMachine` process running in `htop`.

Then I added the extra option `--arch x86_64`, but then it started the VM with `qemu` engine. 
As per lima's [VM types](https://lima-vm.io/docs/config/vmtype/) docs, Rosetta is only supported on `vz` engine.
I got x86_64 Ubuntu, but it is very slow, as it is also mentioned in the lima docs.

Additionally, no `com.apple.Virtualization.VirtualMachine` process was running. In `htop` I could only see a `qemu-system-x86_64` process running. This means that Rosetta was not running?

Another weird thing, running`colima status` on this new VM shows "running using macOS Virtualization.Framework"

```
-> % colima status -p ubuntu
INFO[0000] colima [profile=ubuntu] is running using macOS Virtualization.Framework
INFO[0000] arch: x86_64
INFO[0000] runtime: docker
INFO[0000] mountType: virtiofs
INFO[0000] socket: unix:///Users/maneeshsutar/.colima/ubuntu/docker.sock
```
Although no related process is running on my mac.

Just to compare, here's what we see in `colima status`, when colima is started using command `colima start --arch x86_64` i.e. no options corresponding to rosetta:
```
-> % colima status -p ubuntu-true-x86
INFO[0000] colima [profile=ubuntu-true-x86] is running using QEMU
INFO[0000] arch: x86_64
INFO[0000] runtime: docker
INFO[0000] mountType: sshfs
INFO[0000] socket: unix:///Users/maneeshsutar/.colima/ubuntu-true-x86/docker.sock
```

I want to know if
1. Is it possible to run Ubuntu x86_64 guest VM using Rosetta? Or am I missing something? Does rosetta only work on a aarch64 Ubuntu guest OS?
2. The message in the status "running using macOS Virtualization.Framework", is that a bug?

**Comment by maneesh29s** [comment]
I got answer to my first question in this issue in lima's github
https://github.com/lima-vm/lima/issues/2371

Lima only allows  "aarch64" for VZ.
Thus when I pass `--arch x86_64` colima falls back to `qemu` engine.

**Comment by maneesh29s** [comment]
For the 2nd question, I also found out the reason why `colima status` shows macOS virtualization being used.
Because in colima's `Config.DriverLabel`, the condition only checks if the VMtype is VZ. VMType is set to VZ as it is passed from the command line (`--vm-type vz`). But as mentioned before colima runs `qemu` engine when `--arch x86_64` is also passed as option.

So the condition also needs to check if architecture is not x86_64.

**Comment by abiosoft** [comment]
Qemu is always available, so Colima will fall back to qemu when VZ requirements are not met.

However, I admit the user experience could be better and the behaviour can be more transparent by notifying the user accordingly.

**Reply by maneesh29s** [reply]
Thanks for responding.
I have raised an issue #1151 .
